fly/commands/pipeline_overview.go

```diff
@@ -0,0 +1,147 @@
+package commands
+
+import (
+	"os"
+
+	"github.com/concourse/concourse/atc"
+	"github.com/concourse/concourse/fly/commands/internal/displayhelpers"
+	"github.com/concourse/concourse/fly/rc"
+	"github.com/concourse/concourse/fly/ui"
+	"github.com/fatih/color"
+)
+
+type PipelineOverviewCommand struct {
+	Pipeline string `short:"p" long:"pipeline" required:"true" description:"Get resources in this pipeline"`
+	Json     bool   `long:"json" description:"Print command result as JSON"`
+}
+
+func (command *PipelineOverviewCommand) Execute(args []string) error {
+	pipelineName := command.Pipeline
+
+	target, err := rc.LoadTarget(Fly.Target, Fly.Verbose)
+	if err != nil {
+		return err
+	}
+
+	err = target.Validate()
+	if err != nil {
+		return err
+	}
+
+	var headers []string
+	var resources []atc.Resource
+
+	resources, err = target.Team().ListResources(pipelineName)
+	if err != nil {
+		return err
+	}
+
+	if command.Json {
+		err = displayhelpers.JsonPrint(resources)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+
+	headers = []string{"name", "type", "pinned"}
+	table := ui.Table{Headers: ui.TableRow{}}
+	for _, h := range headers {
+		table.Headers = append(table.Headers, ui.TableCell{Contents: h, Color: color.New(color.Bold)})
+	}
+
+	for _, p := range resources {
+		row := ui.TableRow{}
+		row = append(row, ui.TableCell{Contents: p.Name})
+		row = append(row, ui.TableCell{Contents: p.Type})
+
+		var pinnedColumn ui.TableCell
+		if p.PinnedVersion != nil {
+			pinnedColumn.Contents = ui.PresentVersion(p.PinnedVersion)
+		} else {
+			pinnedColumn.Contents = "n/a"
+		}
+
+		row = append(row, pinnedColumn)
+
+		table.Data = append(table.Data, row)
+	}
+	err = table.Render(os.Stdout, true)
+	var jobs []atc.Job
+
+	jobs, err = target.Team().ListJobs(pipelineName)
+	if err != nil {
+		return err
+	}
+
+	if command.Json {
+		err = displayhelpers.JsonPrint(jobs)
```

Comment:
This never executes when `--json` is used because the of the `return` on line 44. Currently only the resources are printed to stdout, jobs are never printed. 

It might also make sense to put the jobs and resources into one json object so people can do stuff with `jq` like `.jobs.jobName`.

```json
{
  "pipeline_name": "some-pipeline",
  "resources": ...,
  "jobs": ...,
}
```

Reply:
This is a good idea, I'll work on incorporating this

Reply:
This has been fixed, it now returns a json array listing resources as elements, and then the jobs as elements.  Stuffing them into their own object together seems to get a little dicey due to the amount and the data that returns.  I could potentially just create a new struct and map it to that so that:
```suggestion
	type overview struct {
	  pipeline_name string `json:"pipeline_name"`
	  resources []string      `json:"resources"`
	  jobs [] string               `json:"jobs"`
	  }
```

The only pitfall I see with this is the amount of data.  With how I just updated we're getting this, just in a less nested format.  If I did it the above way I personally would consider reducing the amount of data just for consumption purposes.

Thoughts @taylorsilva ?

Reply:
Sorry about the late response!

The amount of data may or may not be a lot, always depends on the pipeline. I'm still in favour of making it easier for people to discover how to massage the data with a tool like jq. I think this would end up being the only command that returns two different objects at once (jobs and resources). I'd prefer to have the data mapped to a struct like you mention.

Reply:
Fair points.  I'll get this fixed up in the next day or so and push for final review.  Thanks @taylorsilva !

Reply:
All set @taylorsilva the json output is now the proposed object with a name attribute, and job/resource attributes that are a slice of their respective type.